### PR TITLE
port influxql docs into 2.5

### PR DESCRIPTION
**Pull request by lwandzura**
Closes https://github.com/influxdata/docs-v2/issues/3417

moved influxql docs into v2.5. - (note that the branch is still called influxql-2.4)

@gunnaraasen please review the supported/unsupported statements and commands to make sure they are accurate

### CODE REVIEW

**Review by sanderson — CHANGES_REQUESTED** [review]
This review is only partially done. Github stopped loading the PR diff, so I'm submitting as-is and will continue on with more review 🤞🏻 .

**Review by sanderson — CHANGES_REQUESTED** [review]
This review is only partially done. Github stopped loading the PR diff, so I'm submitting as-is and will continue on with more review 🤞🏻 .

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.4/query-data/influxql/explore-data/select.md [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+[Frequently Asked Questions](/influxdb/v2.4/reference/faq/#how-does-influxdb-handle-field-type-discrepancies-across-shards)
+document for more information on how InfluxDB handles field value type discrepancies.
+
+### Syntax
+
+```sql
+> SELECT_clause <field_key>::<type> FROM_clause
+```
+
+`type` can be `float`, `integer`, `string`, or `boolean`.
```
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/query-data/influxql/explore-data/select.md:476]: posted this question in the Edge Team channel

File: content/influxdb/v2.4/query-data/influxql/explore-data/select.md [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
+
+The query returns no data as casting a float field value to a string is not yet supported.
+
+{{% /expand %}}
+
+{{< /expand-wrapper >}}
+
+## Merge behavior
+
+In InfluxDB, queries merge [series](/influxdb/v2.4/reference/glossary/#series) automatically.
```
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/query-data/influxql/explore-data/select.md:555]: not sure what to add here, the explanation depicts what merge behavior is

File: content/influxdb/v2.4/query-data/influxql/explore-data/time-and-timezone.md [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+nanosecond UNIX epoch format by default.
+Specify alternative formats with the
+[`precision <format>` command](/influxdb/v2.4/tools/influxql-shell/#precision).  
+
+If you are using the [InfluxQL shell](/influxdb/v2.4/tools/influxql-shell/), use the precision helper command `precision rfc3339` to view results in human readable format.
+
+The [InfluxDB API](/influxdb/v2.4/reference/api/influxdb-1x/) returns timestamps
+in [RFC3339](https://www.ietf.org/rfc/rfc3339.txt) format by default.
+Specify alternative formats with the
+[`epoch` query string parameter](/influxdb/v2.4/reference/api/influxdb-1x/).
```
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/query-data/influxql/explore-data/time-and-timezone.md:37]: added intro to the doc, moved timezone and updated TOC

File: content/influxdb/v2.4/query-data/influxql/explore-data/regular-expressions.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+{{% /influxql/table-meta %}}
+
+| time   | mean |
+| :------------------ | ---------------------:|
+| 1970-01-01T00:00:00Z |  4.4418434585|
+
+This query uses the InfluxQL [MEAN() function](/influxdb/v2.4/query-data/influxql/view-functions/aggregates/#mean) to calculate the average `water_level` across all data that have a tag value for `location`.
+
+#### Use a regular expression to specify a field value in the WHERE clause
+
```
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/query-data/influxql/explore-data/regular-expressions.md:166]: I can add at the top and combine with this:

InfluxQL does not support using regular expressions to match
non-string field values in the `WHERE` clause,
[databases](/influxdb/v2.4/reference/glossary/#database), and
[retention policies](/influxdb/v2.4/reference/glossary/#retention-policy-rp).

File: content/influxdb/v2.4/query-data/influxql/explore-data/select.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/query-data/influxql/explore-data/select.md:476]: no support, confirmed by David B.  Same as 1.x

File: content/influxdb/v2.4/query-data/influxql/explore-data/select.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/v2.4/query-data/influxql/explore-data/select.md:546]: Oh, you're right. The example does explain it. This is fine as-is.

**Pull request merged**